stingray/tests/test_bexvar.py

```diff
@@ -135,3 +135,26 @@ def test_calculate_bexvar(self):
         # Compares lower 1 sigma quantile of the estimated scatter of the log(count rate) in dex 
 
         assert np.isclose(scatt_lo_function, scatt_lo_result, rtol = 0.1)
+
+
+class TestBadValues(object):
+    @classmethod
+    def setup_class(cls):
+        fname_data = os.path.join(datadir, "lcurveA.fits")
+        hdul = fits.open(fname_data)[1]     
+        lightcurve = Table.read(hdul, format = 'fits')
+
+        cls.time = lightcurve["TIME"]
+        time_delta = np.array(list(map(operator.sub, cls.time[1:], cls.time[:-1])))
+        cls.time_delta = np.append(time_delta,float(1.0))
+        cls.src_count = lightcurve["RATE1"]*cls.time_delta
+        cls.frac_exp = lightcurve["FRACEXP"]
+
+    @pytest.mark.skipif("not _HAS_ULTRANEST")
+    def test_weights_sum_warning(self):
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
For these situations, we prefer to use `with pytest.warns(UserWarning) as record:`
See, e.g., https://github.com/StingraySoftware/stingray/blob/main/stingray/tests/test_fourier.py#L133

Reply:
Changed it accordingly.